File: 1833-find-the-highest-altitude/find-the-highest-altitude.py
class Solution(object):
    def largestAltitude(self, gain):
        """
        :type gain: List[int]
        :rtype: int
        """

        arr = []
        n = len(gain)
        arr.append(0)
        mx = 0
        for i in range(n):
            arr.append(arr[i] + gain[i])
            mx = max(mx, arr[i+1])
        
        return mx

        

File: 1833-find-the-highest-altitude/find-the-highest-altitude.cpp
class Solution {
public:
    int largestAltitude(vector<int>& gain) {
        
        int mx = 0;
        int val = 0;
        for(int i=0;i<gain.size();i++){
            mx = max(mx, val + gain[i]);
            val = val + gain[i];
        }

        return mx;
    }
};